src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,116 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private int size;
+    private int capacity;
+    private Object[] elements;
+
+    public ArrayList() {
+        capacity = DEFAULT_CAPACITY;
+        elements = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
+    public ArrayList(int capacity) {
+        this.capacity = capacity;
+        elements = (T[]) new Object[capacity];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size >= capacity) {
+            increaseCapacity();
+        }
+        elements[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        checkIndexForAdd(index);
+        if (size + 1 >= capacity) {
+            increaseCapacity();
+        }
+        System.arraycopy(elements, 0, elements, 1, size);
+        elements[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add((T) list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        return (T) elements[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        checkIndex(index);
+        elements[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        T removed = (T) elements[index];
+        System.arraycopy(elements, index + 1, elements, index, size - index - 1);
+        elements[--size] = null;
+        return removed;
     }
 
     @Override
     public T remove(T element) {
-        return null;
+
+        for (int i = 0; i < size; i++) {
+            if (elements[i] == element || elements[i] != null && elements[i].equals(element)) {
+                return remove(i);
+            }
+
+        }
+        throw new NoSuchElementException();
```

Comment:
Make your exceptions informative.

If you throw an Exception, you should always add an informative message so that it would be clear from the stack trace what exactly went wrong.